src/pages/workspaces/workspace/analysis/modals/AnalysisDuplicator.test.ts

```diff
@@ -0,0 +1,283 @@
+  lastModified: new Date().getTime(),
+  cloudProvider
+})
+
+const baseTestFile: AnalysisFile = getTestFile('test/file0.ipynb' as AbsolutePath)
+
+const onDismiss = jest.fn()
+const onSuccess = jest.fn()
+
+const defaultModalProps: AnalysisDuplicatorProps = {
+  destroyOld: false,
+  fromLauncher: false,
+  printName: baseTestFile.fileName,
+  toolLabel: toolLabelTypes[tools.Jupyter.label],
+  workspaceInfo: defaultGoogleWorkspace.workspace,
+  onDismiss, onSuccess
+}
+
+describe('AnalysisDuplicator', () => {
+  beforeEach(() => {
+    // Arrange
+    const googleStorageMock: Partial<GoogleStorageContract> = ({
+      listAnalyses: jest.fn(() => Promise.resolve([]))
+    })
+    asMockedFn(GoogleStorage).mockImplementation(() => googleStorageMock as GoogleStorageContract)
+
+    const azureStorageMock: Partial<AzureStorageContract> = ({
+      listNotebooks: jest.fn()
+    })
+    asMockedFn(AzureStorage).mockImplementation(() => azureStorageMock as AzureStorageContract)
+
+    asMockedFn(useAnalysisFiles).mockImplementation(() => ({
+      analyses: [],
+      refresh: () => {},
+      loading: false
+    }))
+  })
+
+  it('renders correctly by default with destroyOld false', () => {
+    render(h(AnalysisDuplicator, defaultModalProps))
+
+    screen.getByText(`Copy "${baseTestFile.fileName}"`)
+  })
+
+  it('renders correctly by default with destroyOld true', () => {
+    render(h(AnalysisDuplicator, {
+      ...defaultModalProps,
+      destroyOld: true
+    }))
+
+    screen.getByText(`Rename "${baseTestFile.fileName}"`)
+  })
+
+  it.each(
+    [
+      { inputText: ' ', errorMsg: 'Name can\'t be blank' },
+      { inputText: 'invalid$', errorMsg: 'Name can\'t contain these characters:' },
+      { inputText: 'file1', errorMsg: 'Name already exists' }
+    ]
+  )('rejects existing and invalid names', async ({ inputText, errorMsg }) => {
+    const fileList = [getTestFile('test/file1.ipynb' as AbsolutePath), getTestFile('test/file2.ipynb' as AbsolutePath)]
+    asMockedFn(useAnalysisFiles).mockImplementation(() => ({
+      analyses: fileList,
+      refresh: () => {},
+      loading: false
+    }))
+
+    act(() => {
+      render(h(AnalysisDuplicator, defaultModalProps))
+    })
+
+    const input = screen.getByLabelText(/New Name/)
+    await userEvent.type(input, inputText)
+
+    expect(screen.getAllByText(errorMsg).length).toBe(2)
+    const button = screen.getByText('Copy Analysis')
+    // @ts-expect-error
+    expect(button).toHaveAttribute('disabled')
+  })
+
+  it('copies for a google workspace correctly', async () => {
+    const fileList = [getTestFile('test/file1.ipynb' as AbsolutePath), getTestFile('test/file2.ipynb' as AbsolutePath)]
+    const calledMock = jest.fn()
+    const analysisMock = jest.fn(() => ({
+      copy: calledMock
+    }))
+    const googleStorageMock: Partial<GoogleStorageContract> = ({
+      listAnalyses: () => Promise.resolve(fileList),
+      // @ts-expect-error
```

Comment:
Still discouraged by ignoring TS in mocks. @msilva-broad any ideas for how to resolve this?